cms/datasets/views.py

```diff
@@ -0,0 +1,109 @@
+from collections.abc import Iterable
+from typing import Any
+
+from django import forms
+from django.db.models import QuerySet
+from django.views import View
+from wagtail.admin.forms.choosers import BaseFilterForm
+from wagtail.admin.ui.tables import Column
+from wagtail.admin.views.generic.chooser import (
+    BaseChooseView,
+    ChooseResultsViewMixin,
+    ChooseViewMixin,
+    ChosenResponseMixin,
+    ChosenViewMixin,
+    CreationFormMixin,
+)
+from wagtail.admin.viewsets.chooser import ChooserViewSet
+
+from cms.datasets.models import Dataset, ONSDataset
+
+
+class DatasetBaseChooseViewMixin:
+    @property
+    def columns(self):  # type: ignore
+        return [
+            *getattr(super(), "columns", []),
+            Column("edition", label="Edition", accessor="formatted_edition"),
+            Column("version", label="Latest Version", accessor="version"),
+        ]
+
+
+class DatasetSearchFilterForm(BaseFilterForm):
+    q = forms.CharField(
+        label="Search datasets",
+        widget=forms.TextInput(attrs={"placeholder": "Dataset title"}),
+        required=False,
+    )
+
+    def filter(self, objects: Iterable[Any]) -> Iterable[Any]:
+        objects = super().filter(objects)
+        search_query = self.cleaned_data.get("q")
+        if search_query:
+            search_query_lower = search_query.strip().lower()
+            objects = [obj for obj in objects if self.obj_matches_search_query(obj, search_query_lower)]
+            self.is_searching = True
+            self.search_query = search_query
+        return objects
+
+    @staticmethod
+    def obj_matches_search_query(obj: Any, search_query_lower: str) -> bool:
+        return any(search_query_lower in getattr(obj, search_field).lower() for search_field in obj.search_fields)
+
+
+class ONSDatasetBaseChooseView(BaseChooseView):
+    model_class = ONSDataset
+    filter_form_class = DatasetSearchFilterForm
+
+    def get_object_list(self) -> QuerySet[ONSDataset]:
+        # Due to pagination this is required for search to check entire list
+        # The hardcoded limit will need changing.
+        object_list: QuerySet[ONSDataset] = self.model_class.objects.filter(limit=1000)  # pylint: disable=no-member
```

Comment:
Don't think this is needed anymore. `get_object_list` on `BaseChooseView` returns all objects so should work as expected. Changes you made elsewhere likely made this redundant.